lambda-buffers-compiler/src/LambdaBuffers/Compiler/TypeClass/Rules.hs

```diff
@@ -5,16 +5,22 @@ module LambdaBuffers.Compiler.TypeClass.Rules (
   Class (..),
   Constraint (..),
   Rule (..),
-  type Instance,
+  FQClassName (..),
   mapPat,
+  ruleHeadPat,
+  ruleHeadClass,
 ) where
 
-import LambdaBuffers.Compiler.ProtoCompat qualified as P
+import Data.Text (Text)
+
 import LambdaBuffers.Compiler.TypeClass.Pat (Pat)
 
+data FQClassName = FQClassName {cName :: Text, cModule :: [Text]}
+  deriving stock (Show, Eq, Ord)
+
 data Class = Class
-  { name :: P.TyClassRef
-  , supers :: [Class]
+  { cname :: FQClassName
+  , csupers :: [Class]
   }
```

Comment:
Same comment as above. No need to change it, but consider the alternative approach with context building.

Reply:
I'm indifferent here, what you're suggesting would work fine but would introduce the possibility of a new error type when we look up a class that should exist. There are already enough of those in `Utils` in the other PR.... but also we effectively construct the context you're proposing in order to construct all the classes so /shrug. 

I'm probably going to keep it as-is for now because it makes `solve` simpler but like I said, i have no strong feelings about this